File: app/src/main/java/com/anokmik/tripassistant/model/ObservableTripEvent.java
package com.anokmik.tripassistant.model;

import android.databinding.BaseObservable;

import com.anokmik.persistence.model.TripEvent;

public final class ObservableTripEvent extends BaseObservable {

    private TripEvent tripEvent;

    public ObservableTripEvent(TripEvent tripEvent) {
        this.tripEvent = tripEvent;
    }

    public TripEvent get() {
        return tripEvent;
    }

    public void set(TripEvent tripEvent) {
        this.tripEvent = tripEvent;
        notifyChange();
    }

    public String getName() {
        return tripEvent.name;
    }

    public void setName(String name) {
        tripEvent.name = name;
    }

    public String getComment() {
        return tripEvent.comment;
    }

    public void setComment(String comment) {
        tripEvent.comment = comment;
    }

    public long getStartDate() {
        return tripEvent.startDate;
    }

    public void setStartDate(long startDate) {
        tripEvent.startDate = startDate;
        notifyChange();
    }

    public long getFinishDate() {
        return tripEvent.finishDate;
    }

    public void setFinishDate(long finishDate) {
        tripEvent.finishDate = finishDate;
        notifyChange();
    }

    public void save() {
        tripEvent.save();
    }

    public void delete() {
        tripEvent.delete();
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/model/ObservableTripEvent.kt
package com.anokmik.tripassistant.model

import android.databinding.BaseObservable

import com.anokmik.persistence.model.TripEvent

class ObservableTripEvent(private var tripEvent: TripEvent?) : BaseObservable() {

    fun get(): TripEvent? {
        return tripEvent
    }

    fun set(tripEvent: TripEvent?) {
        this.tripEvent = tripEvent
        notifyChange()
    }

    var name: String?
        get() = tripEvent?.name
        set(name) {
            tripEvent?.name = name
        }

    var comment: String?
        get() = tripEvent?.comment
        set(comment) {
            tripEvent?.comment = comment
        }

    var startDate: Long?
        get() = tripEvent?.startDate
        set(startDate) {
            tripEvent?.startDate = startDate
            notifyChange()
        }

    var finishDate: Long?
        get() = tripEvent?.finishDate
        set(finishDate) {
            tripEvent?.finishDate = finishDate
            notifyChange()
        }

    fun save() = tripEvent?.save()

    fun delete() = tripEvent?.delete()

}